### PR TITLE
fix(dance): correct IoU normalization, make tutorial learn, trim API

### PR DESCRIPTION
**Pull request by jarod1212**
Thanks for kicking this off, @bruAristimunha ! It's now fully working end-to-end. On top of your branch I changed three things:

- **IoU loss** (`training/losses.py`): it was averaged over all queries, so unmatched no-object slots each added `1 − 0 = 1` and washed out the localization signal. Now normalized over the Hungarian-matched spans only — this is what lets the event F1 move off zero.
- **Tutorial** (`plot_dance_event_detection.py`): reworked the synthetic task so it visibly learns (F1-event 0.00 → ~0.35), plus CPU/GPU handling.
- **API trim**: folded `detr_to_dense_probs` into `DanceLoss` and dropped `events_to_mask`, `extract_events_from_detr_batch` and `f1_sample` (the tutorial scores the dense head with torchmetrics). `whats_new.rst` updated to match.

25 DANCE tests + pre-commit all green. Tell me if I should change anything or if something's unclear. Thanks again!

### CODE REVIEW

**Review by bruAristimunha — APPROVED** [review]
Super good @jarod1212, once everything is green, I am happy to merge, as you are the author of the model

File: examples/applied_examples/plot_dance_event_detection.py [hunk 1]
```diff
@@ -229,45 +259,60 @@ def annotations_to_events(raw, n_classes):
...
 # %%
-# Train a few steps
-# -----------------
+# Train the model
+# ---------------
+# A short run on this small synthetic set is enough to see the event F1 climb off
+# zero; real recordings need a full training schedule.
 model.train()
-for epoch in range(2):
+for epoch in range(120):
```
Comment (bruAristimunha) [inline examples/applied_examples/plot_dance_event_detection.py]: We need to reduce for the CI the epoch, or load the weight, only detail to change

**Review by bruAristimunha — COMMENTED** [review]
Small detail

File: examples/applied_examples/plot_dance_event_detection.py (hunk 1 above)
Comment (jarod1212) [reply examples/applied_examples/plot_dance_event_detection.py]: Done — defaulted to 2 epochs with a comment pointing to n_epochs = 120 for a full run. Thanks!

**Comment by jarod1212** [comment]
Noted! I think you need to "Approve and run workflows" as it's my first PR. I've run pre-commit and the DANCE tests are all green. Thanks @bruAristimunha !

**Comment by bruAristimunha** [comment]
Many thanks for the PR @jarod1212, i hope to see you more here :)
